File: project-euler.java/src/main/java/euler/solutions/_3.java
package euler.solutions;

import euler.common.BaseSolution;

public class _3 extends BaseSolution {

    @Override
    public long solve() {
        long num = 600851475143L;
        int  fac = 2;
        while (fac < num) {
            if (num % fac == 0) {
                num /= fac;
            }
            else {
                fac++;
            }
        }
        return num;
    }
}

File: project-euler.go/solutions/_3/3.go
package _3

func Solve() int64 {
	var num int64 = 600851475143
	var fac int64 = 2
	for fac < num {
		if num % fac == 0 {
			num /= fac
		} else {
			fac++
		}
	}
	return num
}
